### PR TITLE
Replacing the exclusionary words whitelist and blacklist that won't impact the compatibility. 

### PR DESCRIPTION
**Pull request by aponb**
### Description
Replacing the exclusionary words whitelist and blacklist that won't impact the compatibility. Changes just in code comments, internal variables, methods and class names.
 
### Issues Resolved
This Pull request replaces Pull request #744 
 
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Signed-off-by: Andreas Pre aponb@gmx.at

### CODE REVIEW

**Comment by nknize** [comment]
Thx @aponb.. looks like a legit comparison failure. I could repro locally:

```
REPRODUCE WITH: ./gradlew ':modules:reindex:test' --tests "org.opensearch.index.reindex.ReindexFromRemoteWhitelistTests.testUnwhitelistedRemote" -Dtests.seed=95E696791CA7DA1F -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=sv-SE -Dtests.timezone=Australia/Melbourne -Druntime.java=17
```

```
org.opensearch.index.reindex.ReindexFromRemoteWhitelistTests > testUnwhitelistedRemote FAILED
    org.junit.ComparisonFailure: expected:<...ist:1344494910] not [white]listed in reindex.re...> but was:<...ist:1344494910] not [allow]listed in reindex.re...>
        at __randomizedtesting.SeedInfo.seed([95E696791CA7DA1F:C249AF47E210FE99]:0)
        at org.junit.Assert.assertEquals(Assert.java:117)
        at org.junit.Assert.assertEquals(Assert.java:146)
        at org.opensearch.index.reindex.ReindexFromRemoteWhitelistTests.testUnwhitelistedRemote(ReindexFromRemoteWhitelistTests.java:134)
```

File: modules/reindex/src/test/java/org/opensearch/index/reindex/ReindexFromRemoteWhitelistTests.java [hunk 1]
```diff
@@ -104,32 +104,32 @@ public void testWhitelistedByPrefix() {
...
     public void testUnwhitelistedRemote() {
         int port = between(1, Integer.MAX_VALUE);
-        List<String> whitelist = randomBoolean() ? randomWhitelist() : emptyList();
+        List<String> allowlist = randomBoolean() ? randomAllowlist() : emptyList();
         Exception e = expectThrows(
             IllegalArgumentException.class,
-            () -> checkRemoteWhitelist(buildRemoteWhitelist(whitelist), newRemoteInfo("not in list", port))
+            () -> checkRemoteAllowlist(buildRemoteAllowlist(allowlist), newRemoteInfo("not in list", port))
         );
         assertEquals("[not in list:" + port + "] not whitelisted in reindex.remote.whitelist", e.getMessage());
```
Comment (tlfeng) [inline modules/reindex/src/test/java/org/opensearch/index/reindex/ReindexFromRemoteWhitelistTests.java:134]: Thank @nknize pointing out the reason to the test failure, seems the `whitelisted` is missed.

**Comment by tlfeng** [comment]
I really appreciate you for taking the time in this PR! 🏆 Especially in picking out the non-inclusive terms that doesn't break any compatibility at all.

**Comment by tlfeng** [comment]
Ah, please use `git commit --amend -s` to append the DCO sign-off to your last commit. Github Website didn't add the sign-off.

**Comment by aponb** [comment]
> About test log 2097, the format check failure is a known issue in current `main` branch. #1994
> 
> Update: The existing test failure has been fixed in commit [054595b](https://github.com/opensearch-project/OpenSearch/commit/054595b140e236d1a3e0fdbc93de58bb22f7d350). I think there will be no issue after:
> 
>     1. `git commit --amend -s` to add the DCO sign-off to your last commit of "merge"
> 
>     2. Merge the current `main` branch into your branch, with DCO sign-off
> 
> 
> Thank you! grin

Done. Hope that was ok!

**Comment by tlfeng** [comment]
Thanks @aponb ! The DCO check is fine now. Only the unit test failure left.

**Comment by tlfeng** [comment]
Hi @aponb, when you get a time, could you resolve the remaining small issues in the Pull Request? So that it can be merged.

1. There are 2 commits doesn’t have [DCO sign-off](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin). It’s a bit troublesome, you need either squash your commits in the PR into one, or modify the 2 commits in the middle.
2. A unit test failure caused by L134 in `ReindexFromRemoteWhitelistTests.java`

Feel free to tell us if you need any help. Thank you! 👍

**Comment by tlfeng** [comment]
Hi @aponb ,

To fix the DCO sign-off check, I'd like to provide a solution:
1. Backup your current branch with all the changes in the Pull Request, into a new branch, such as named "backup_whitelist_blacklist":  `git checkout -b backup_whitelist_blacklist`
2. Follow the solution here to squash all commits on your branch to overwrite the commits that doesn’t have sign-off message: https://stackoverflow.com/a/50880042/11408564. Since your changes are in `main` branch, you need a slightly different command:
```
git reset --soft origin/main
# Add files for the commit.
git add .
git commit -m "Replacing the exclusionary word whitelist that won't impact the compatibility. -s
# force push the commit to your remote repo
git push —force
```

In addition, I suggest use a new branch other than the default `main` branch to create Pull Requests.

> because any new changes you add to the master branch will automatically show up in the pull request. That means changes you did not intend to be there, would now be there.

(source: https://blog.jasonmeridth.com/posts/do-not-issue-pull-requests-from-your-master-branch/)

**Comment by tlfeng** [comment]
Another small test failure:
Looks like the `whitelisted` needs to be changed in https://github.com/opensearch-project/OpenSearch/blob/1.2.4/modules/reindex/src/yamlRestTest/resources/rest-api-spec/test/reindex/20_validation.yml#L311
```
> Task :modules:reindex:yamlRestTest

REPRODUCE WITH: ./gradlew ':modules:reindex:yamlRestTest' --tests "org.opensearch.index.reindex.ReindexClientYamlTestSuiteIT.test {yaml=reindex/20_validation/unwhitelisted remote host fails}" -Dtests.seed=35B7ED975A16E179 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=nb-NO -Dtests.timezone=America/Anchorage -Druntime.java=17

org.opensearch.index.reindex.ReindexClientYamlTestSuiteIT > test {yaml=reindex/20_validation/unwhitelisted remote host fails} FAILED
    java.lang.AssertionError: Failure at [reindex/20_validation:310]: the error message was expected to match the provided regex but didn't
    Expected: \[badremote:9200\] not whitelisted in reindex.remote.whitelist
         but: was "{root_cause=[{type=illegal_argument_exception, reason=[badremote:9200] not allowlisted in reindex.remote.whitelist, stack_trace=[[badremote:9200] not allowlisted in reindex.remote.whitelist]; nested: IllegalArgumentException[[badremote:9200] not allowlisted in reindex.remote.whitelist];\n\tat 
```

@aponb Thanks a lot for your rapid response! 👍

**Comment by aponb** [comment]
> Another small test failure: Looks like the `whitelisted` needs to be changed in https://github.com/opensearch-project/OpenSearch/blob/1.2.4/modules/reindex/src/yamlRestTest/resources/rest-api-spec/test/reindex/20_validation.yml#L311
> 
> ```
> > Task :modules:reindex:yamlRestTest
> 
> REPRODUCE WITH: ./gradlew ':modules:reindex:yamlRestTest' --tests "org.opensearch.index.reindex.ReindexClientYamlTestSuiteIT.test {yaml=reindex/20_validation/unwhitelisted remote host fails}" -Dtests.seed=35B7ED975A16E179 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=nb-NO -Dtests.timezone=America/Anchorage -Druntime.java=17
> 
> org.opensearch.index.reindex.ReindexClientYamlTestSuiteIT > test {yaml=reindex/20_validation/unwhitelisted remote host fails} FAILED
>     java.lang.AssertionError: Failure at [reindex/20_validation:310]: the error message was expected to match the provided regex but didn't
>     Expected: \[badremote:9200\] not whitelisted in reindex.remote.whitelist
>          but: was "{root_cause=[{type=illegal_argument_exception, reason=[badremote:9200] not allowlisted in reindex.remote.whitelist, stack_trace=[[badremote:9200] not allowlisted in reindex.remote.whitelist]; nested: IllegalArgumentException[[badremote:9200] not allowlisted in reindex.remote.whitelist];\n\tat 
> ```
> 
> @aponb Thanks a lot for your rapid response! +1

Thanks! This is fixed now.

**Comment by aponb** [comment]
> Hi @aponb ,
> 
> To fix the DCO sign-off check, I'd like to provide a solution:
> 
>     1. Backup your current branch with all the changes in the Pull Request, into a new branch, such as named "backup_whitelist_blacklist":  `git checkout -b backup_whitelist_blacklist`. In case your current code change is lost by mistake during the following commands
> 
>     2. Follow the solution here to squash all commits on your branch to overwrite the commits that doesn’t have sign-off message: https://stackoverflow.com/a/50880042/11408564. Since your changes are in `main` branch, you need a slightly different command:
> 
> 
> ```
> # checkout the branch in this Pull Request
> git checkout main
> # Remove all your new commits, while remain your code change.
> git reset --soft origin/main
> # Add files for the commit.
> git add .
> git commit -s -m "Replacing the exclusionary word whitelist that won't impact the compatibility."
> # force push the commit to your remote repo
> git push —force
> ```
> 
> In addition, I suggest use a new branch other than the default `main` branch to create Pull Requests.
> 
> > because any new changes you add to the master branch will automatically show up in the pull request. That means changes you did not intend to be there, would now be there.
> 
> (source: https://blog.jasonmeridth.com/posts/do-not-issue-pull-requests-from-your-master-branch/)

I followed your suggestion. It seems it worked now. Thanks for your help and patience!

**Comment by tlfeng** [comment]
Hi @aponb, you are very welcome! Thanks to your patience as well.
Hmmm, seems I made a small mistake on the command `git reset --soft origin/main` in my suggestion. Sorry that my solution was not comprehensive.
DCO sign-off is totally fine, but there are many unrelated commits checked in the PR, which must caused by the `git reset` command...
Only the 4 commits should be shown in the PR:
https://github.com/opensearch-project/OpenSearch/pull/1970/commits/f03dcaba52f4ed33a97d1031ba95cf259a6ff3f4 https://github.com/opensearch-project/OpenSearch/pull/1970/commits/002147537d64a90d5a15e66e24f479619b0820bf https://github.com/opensearch-project/OpenSearch/pull/1970/commits/86157af85251152be290dbe3b85c50166ce98245 https://github.com/opensearch-project/OpenSearch/pull/1970/commits/703309af2b011921f10d29499f84fc3c47f8f276

Please keep your backup branch (https://github.com/aponb/OpenSearch/tree/backup_whitelist_blacklist3), let me find a way to deal with it.

**Comment by tlfeng** [comment]
Hi @aponb, finally I got a solution to our current situation 😁.
In this way, you will need to raise a new Pull Request based on your feature branch instead the current `main` branch, and we will merge your code changes in the new PR.
1. Make sure the branch `backup_whitelist_blacklist3` is safe.
2. Update the current `main` branch of your fork repository sync with the upstream repository (https://github.com/opensearch-project/OpenSearch)
Reference: https://stackoverflow.com/a/42332860/11408564
```
# Add a remote repository of the upstream repository if there is not one locally
git remote add upstream git@github.com:opensearch-project/OpenSearch.git

# ensures current branch is main
git checkout main

# fetch all new commits made to upstream/main
git fetch upstream main

# Reset your local main branch, and this will delete all your local changes to main
git reset --hard upstream/main

# take care, this will delete all your changes on your forked main branch
git push origin main --force
```
3. Use a new feature branch to hold your code changes
```
# ensures current branch is main
git checkout main 

# Cut a new branch from main branch, whatever the new branch name is.
git checkout -b replace-whitelist-blacklist

# Cherry pick the 4 commits of your actual code changes
git cherry-pick f03dcaba52f4ed33a97d1031ba95cf259a6ff3f4 002147537d64a90d5a15e66e24f479619b0820bf 86157af85251152be290dbe3b85c50166ce98245 703309af2b011921f10d29499f84fc3c47f8f276

# Push the feature branch to your remote repository
git push --set-upstream origin replace-whitelist-blacklist
```
4. Create a new Pull Request based on the feature branch

Thanks for your time!

**Comment by aponb** [comment]
> Hi @aponb, finally I got a solution to our current situation grin. In this way, you will need to raise a new Pull Request based on your feature branch instead the current `main` branch, and we will merge your code changes in the new PR.
> 
>     1. Backup your current `main` branch by creating a new branch based on `main`.
> 
> 
> ```
> # ensure current branch is main
> git checkout main
> 
> # cut a new branch based on main branch as the backup, whatever the new branch name is
> git checkout -b backup-whitelist-blacklist
> ```
> 
>     2. Sync up the current `main` branch of your fork repository with the `main` branch of upstream repository (https://github.com/opensearch-project/OpenSearch)
>        Reference: https://stackoverflow.com/a/42332860/11408564
> 
> 
> ```
> # Add a remote repository of the upstream repository, if there is not one locally
> git remote add upstream git@github.com:opensearch-project/OpenSearch.git
> 
> # ensure current branch is main
> git checkout main
> 
> # fetch all new commits made to upstream/main
> git fetch upstream main
> 
> # Reset your local main branch, and this will delete all your local changes to main
> git reset --hard upstream/main
> 
> # take care, this will delete all your changes on your forked main branch
> git push origin main --force
> ```
> 
>     3. Use a new feature branch to hold your code changes
> 
> 
> ```
> # ensures current branch is main
> git checkout main 
> 
> # Cut a new branch from main branch, whatever the new branch name is.
> git checkout -b replace-whitelist-blacklist
> 
> # Cherry pick the 4 commits of your actual code changes
> # (or running `git cherry-pick <hash of the commit>` separately for each commit)
> git cherry-pick f03dcaba52f4ed33a97d1031ba95cf259a6ff3f4 002147537d64a90d5a15e66e24f479619b0820bf 86157af85251152be290dbe3b85c50166ce98245 703309af2b011921f10d29499f84fc3c47f8f276
> 
> # Push the feature branch to your remote repository
> git push --set-upstream origin replace-whitelist-blacklist
> ```
> 
>     4. Create a new Pull Request based on the feature branch
> 
> 
> Thanks for your time!

Thanks for your effort. Please see https://github.com/opensearch-project/OpenSearch/pull/2178

**Comment by tlfeng** [comment]
Hi @aponb, I'm so glad that we finally made it! 🎉 🤝

**Comment by aponb** [comment]
> Hi @aponb, I'm so glad that we finally made it! tada handshake

Thanks for your help and your cooperation. I learned a lot, which will hopefully helpful in future!

**Pull request closed without merging**